ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/JarManifestErrorReporter.java

```diff
@@ -139,17 +141,32 @@ protected void parseManifest(IDocument document, IProgressMonitor monitor) {
 				}
 				// parse
 				if (line.length() == 0) {
-					// Empty Line
+					// Empty Line in the beginning
 					if (l == 0) {
 						report(PDECoreMessages.BundleErrorReporter_noMainSection, 1, CompilerFlags.ERROR, PDEMarkerFactory.CAT_FATAL);
 						return;
+
+						// empty line elsewhere
+					} else if (l != document.getNumberOfLines() - 1) {
+						emptyLines.add(Integer.valueOf(l));
+						continue;
 					}
 					/* flush last line */
 					if (header != null) {
 						fHeaders.put(header.getName().toLowerCase(), header);
 						header = null;
 					}
 					break; /* done processing main attributes */
+				} else {
+					for (Integer index : emptyLines) {
+						VirtualMarker marker = report(PDECoreMessages.BundleErrorReporter_noNameHeader, index + 1,
+								CompilerFlags.WARNING, PDEMarkerFactory.M_EXTRANEOUS_EMPTY_LINES,
```

Comment:
Why do we use `CompilerFlags.WARNING` here if we are sure and report user that the manifest is invalid?

Reply:
we initially started with `CompilerFlags.ERROR`, but then downplayed it to warning due to the regression it caused. Is it ok to see its stability for sometime before we elevate it to an error?

Reply:
> then downplayed it to warning due to the regression it caused

Well, if we **know** the manifest is invalid, why do we want to report only a **warning**?

Regarding regression: we should only make sure the code does the right thing :-)

Reply:
@vik-chand - pls share your thoughts here. Error or Warning? I don't have a strong opinion at this point, so would go by the consensus between you two!

Reply:
Lets keep to whatever was before ( ie Error)

Reply:
thanks @vik-chand and @iloveeclipse . pushed (and squashed) a change to that effect, pls have a look. thanks!